### ISSUE TITLE
Out-of-order partitions can prompt constantly new empty partitions

### ISSUE THREAD

**Issue by jcjones** [letsencrypt/mariadb-sequential-partition-manager-py#35]
If `get_position_increase_per_day` skips a partition because of ordering, we can get a very incorrect rate of change. That can prompt `plan_partition_changes` to predict Start-of-fill far too early, and result in repeatedly triggering a rename of partitions when unnecessary. This should be mostly resolved by the resolution of #30.

**Comment by pgporada** [comment]
This manifests as
```
> SELECT TABLE_NAME, PARTITION_NAME, INDEX_LENGTH, DATA_FREE, CREATE_TIME, UPDATE_TIME FROM information_schema.partitions WHERE TABLE_SCHEMA='database' AND TABLE_NAME = 'fqdnSets' AND PARTITION_NAME IS NOT NULL;
+------------+----------------+--------------+-----------+---------------------+---------------------+
| TABLE_NAME | PARTITION_NAME | INDEX_LENGTH | DATA_FREE | CREATE_TIME         | UPDATE_TIME         |
+------------+----------------+--------------+-----------+---------------------+---------------------+
| fqdnSets   | p_20210817     |        32768 |         0 | 2021-12-16 08:55:16 | NULL                |
| fqdnSets   | p_20210916     |        32768 |   7340032 | 2021-12-16 08:55:16 | NULL                |
| fqdnSets   | p_20211016     |        32768 |   4194304 | 2021-12-16 08:55:16 | NULL                |
| fqdnSets   | p_20211115     |   3438182400 |   6291456 | 2021-12-16 08:55:16 | NULL                |
| fqdnSets   | p_20211215     |   3101622272 |   6291456 | 2021-12-16 08:55:16 | NULL                |
| fqdnSets   | p_20210922     |  33995784192 |   7340032 | 2021-12-16 08:55:16 | 2021-12-16 04:19:07 |
| fqdnSets   | p_20211216     |    228573184 |   7340032 | 2021-12-16 08:55:16 | 2021-12-16 18:35:01 |
| fqdnSets   | p_20211217     |        32768 |         0 | 2021-12-16 08:55:16 | 2021-12-16 18:35:01 |
| fqdnSets   | p_20211218     |        32768 |         0 | 2021-12-16 08:55:16 | 2021-12-16 18:35:01 |
| fqdnSets   | p_20211219     |        32768 |         0 | 2021-12-16 08:55:16 | 2021-12-16 18:35:01 |
| fqdnSets   | p_20211220     |        32768 |         0 | 2021-12-16 08:55:16 | 2021-12-16 18:35:01 |
+------------+----------------+--------------+-----------+---------------------+---------------------+
11 rows in set (0.001 sec)
```

**Comment by jcjones** [comment]
This appears to be fixed by #37 .

**Comment by jcjones** [comment]
Fixed by #37
